### DISCUSSION TITLE
Postanalysis and exportplots to pdf but something went wrong.

### DISCUSSION THREAD

**Discussion by fookloy** [milaboratory/mixcr#1566] [Q&A] [answered]
Hi, i tried to export the cdr3metrics related plots, but could just output an plot.pdf without actual content, anything wrong with my code?

`java -jar /mnt/C_NGS01/Fred/antibody/mixcr.jar postanalysis individual --default-downsampling none --default-weight-function read --chains IGH /mnt/C_NGS01/Fred/antibody/PCRseq/mouse1/HHHCDR3/H-HH_L1_assemble.clns /mnt/C_NGS01/Fred/antibody/PCRseq/mouse1/HHHCDR3/postanalysis.json`

`java -jar /mnt/C_NGS01/Fred/antibody/mixcr.jar exportPlots cdr3metrics --chains IGH /mnt/C_NGS01/Fred/antibody/PCRseq/mouse1/HHHCDR3/postanalysis.json /mnt/C_NGS01/Fred/antibody/PCRseq/mouse1/HHHCDR3/postanalysis.pdf 
`

and the output were below.
Thanks!

[postanalysis.IGH.pdf](https://github.com/milaboratory/mixcr/files/14445608/postanalysis.IGH.pdf)
[postanalysis.json](https://github.com/milaboratory/mixcr/files/14445609/postanalysis.json)

**Comment by mizraelson** [comment]
Hi, what version of MiXCR do you use?

**Reply by fookloy** [reply]
Hi, MIXCR 4.5.0 was used in process, and the code to output .clns file is below:

`java --jar mixcr.jar align --preset generic-amplicon --species human --rna --rigid-left-alignment-boundary --rigid-right-alignment-boundary C -OmergerParameters.minimalOverlap=10 -OmergerParameters.minimalIdentity=0.9 ${id}_1.fq.gz ${id}_2.fq.gz ${id}_align.vdjca`

`java --jar mixcr.jar assemble --assemble-clonotypes-by "{FR1Begin:FR4End}"  -ObadQualityThreshold=0 ${id}_align.vdjca ${id}_assemble.clns`

**Reply by mizraelson** [reply]
Can you please share the `.clns` file.

**Reply by fookloy** [reply]
Sure, github has 40M upload limit, so i wrote an email to the support@milaboratories.com, please check the attachment.
And if you can't see that mail there, please let me know, i will try another way.

**Reply by mizraelson** [reply]
I didn't get your e-mail, unfortunately.

**Reply by fookloy** [reply]
I forwarded it to your personal email, please have a check~

**Reply by mizraelson** [reply, marked as answer]
Can you please try the [latest develop version](https://link.milaboratory.com/software/mixcr/mixcr-develop.zip). I get the result attached.
[cdr3.IGH.pdf](https://github.com/milaboratory/mixcr/files/14491550/cdr3.IGH.pdf)

**Reply by fookloy** [reply]
I tried that latest version, but still output an empty plot.pdf with just the axis, same as yours attachment.
![image](https://github.com/milaboratory/mixcr/assets/156155685/70685bd1-39b5-4aa5-8edc-22faa4067612)

**Reply by mizraelson** [reply]
but the plot is not empty, you have only on sample hence you get a line instead of a boxplot. Each line represents this sample. For instance: length of CDR3 is 47.6

**Reply by fookloy** [reply]
Oh! i see, sorry for the misunderstanding, i thought it would be looks like the plot i attached. 
I used to use the VDJTOOLS to output the cdr3 related plots like the attachment, but it seems as MIXCR updated, VDJTOOLS was no longer support the tsv file type that MIXCR output, and i found a discussion days ago saying that VDJTOOLS functions have already been integrated into MIXCR.
If so , could you please show how to export the attachment-like plot?  And it is OK if no, i would figure it out in ggplot, thanks~




![basic-fancyspectra](https://github.com/milaboratory/mixcr/assets/156155685/769bf4b6-0654-4ec1-a939-7cf0d0946661)

**Reply by mizraelson** [reply]
I see. Yeah, currently we do not have such plot but we will implement it in the future. Also, if you manually rename the columns in the MiXCR output, you can run vdjtools on it.

**Reply by fookloy** [reply]
That's a good way, i'll try, and thank you so much for your help!

**Reply by mizraelson** [reply]
#1300 
This discussion might help with manual converting to VDJtools format

**Reply by fookloy** [reply]
Yes, i found that discussion early this day, that's really detailed and i'll try it later.

And really appreciated for your prompt reply, thanks!
